### DISCUSSION TITLE
Username field not allowing email address for Microsoft Entra ID registration

### DISCUSSION THREAD

**Discussion by ryanr-gpn** [OrchardCMS/OrchardCore#16114] [Q&A]
I'm just getting started. I followed the guide to [Create a Decoupled CMS site](https://docs.orchardcore.net/en/latest/guides/decoupled-cms/). I then followed the guide [Microsoft Entra ID (Azure AD) integration](https://docs.orchardcore.net/en/latest/guides/microsoft-entra-id-integration/) because I want to secure the entire site behind Azure AD. When I try to login/register with my Microsoft Entra ID account, it authenticates properly but when it tries to create the account I get an error that the user name (which is my Microsoft email) is invalid (can only contain letters or digits). 
![image](https://github.com/OrchardCMS/OrchardCore/assets/34683186/adfa83af-d5a9-4699-bd1d-d861941fe3e2)
I also had an issue during initial setup where it wouldn't let me use my gmail address for the username and email. Got the same error so I just used "admin" for the username so I could proceed.

Anyone else have this issue or ideas on how to solve it?

**Comment by douwinga** [comment]
I don't think you can have an '@' in your username anymore. You should be able to fix this with a user registration script
![image](https://github.com/OrchardCMS/OrchardCore/assets/11961383/33e836db-741b-4432-8c84-a8529adcbeb7)

Something like this to just change @ to +:
![image](https://github.com/OrchardCMS/OrchardCore/assets/11961383/8542435f-b645-4aab-9f34-d5747ce88205)
